include/wlr/types/wlr_keyboard_group.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_KEYBOARD_GROUP_H
+#define WLR_TYPES_WLR_KEYBOARD_GROUP_H
+
+#include <wayland-server-core.h>
+#include "wlr/types/wlr_keyboard.h"
+#include "wlr/types/wlr_input_device.h"
+
+struct wlr_keyboard_group {
+	struct wlr_keyboard keyboard;
+	struct wlr_input_device *input_device;
+	struct wl_list devices; // keyboard_group_device::link
+	struct wl_list keys; // keyboard_group_key::link
```

Comment:
Hmm. We already have logic for this in the seat I think.

Reply:
It's possible to have multiple keyboard groups per seat. That's up to the compositor. For example, in my sway PR, there is the ability to group by keymap. Say there are four keyboard devices than span across two keymaps, there would be two keyboard groups for the seat

Reply:
Yeah. My question is: why do we need this extra state? I doesn't seem used anywhere here.

Reply:
I don't think `util/array.h` is sufficient because we need a counter for the number of keyboard where the key is currently pressed The key press for the group should occur on the first key press, but the key release should not occur until ALL keyboards have released the key. Simply adding the key code to a 1-dimensional uint32_t array/set does not allow for this to happen.

Reply:
Ah, you're right!